tools/go.mod

```diff
@@ -4,7 +4,7 @@ go 1.22
 
 require (
 	github.com/golang/mock v1.6.0
-	github.com/golangci/golangci-lint v1.56.2
+	github.com/golangci/golangci-lint v1.57.0
```

Comment:
note: these upgrades introduce some new issues that will need fixed before this PR is merged.
Running `make check-linter` from the branch will show them.